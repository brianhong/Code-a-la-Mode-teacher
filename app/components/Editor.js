// @flow
import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import brace from 'brace';
import AceEditor from 'react-ace';
// import Resizable from 'react-resizable-box';
import 'brace/mode/javascript';
import 'brace/theme/solarized_dark';
import styles from './Home.css';
import FiletreeContainer from '../containers/FiletreeContainer';
import GitControlsContainer from '../containers/GitControlsContainer';
import { ipcRenderer } from 'electron';
import Flexbox from 'flexbox-react';

// ...

type nextPropsType = {
  contents: string,
  changeEditor: () => void
};

class Editor extends Component {
  props: {
    contents: string,
    changeEditor: () => void
  }

  componentWillReceiveProps(nextProps: nextPropsType) {
    ipcRenderer.send('editor-changes', nextProps.contents);
  }

  render() {
    const { changeEditor } = this.props;

    return (
      <Flexbox flexDirection="row" minHeight="100vh" flexWrap="wrap">
        <Flexbox element="header" height="7vh" width="100vw">
          <Link to="/">
            <i className="fa fa-arrow-left fa-3x" />
          </Link>
          <GitControlsContainer />
        </Flexbox>
        <Flexbox flexGrow={1}>
          {/*<Resizable width={'100%'} height={'100%'}>*/}
            <FiletreeContainer className={styles.container} />
          {/*</Resizable>*/}
        </Flexbox>
        <Flexbox flexGrow={4}>
          {/*<Resizable width={'100%'} height={'100%'}>*/}
            <AceEditor
              mode="javascript"
              theme="solarized_dark"
              fontSize={15}
              width={"100%"}
              height={"100%"}
              onChange={changeEditor}
              name="UNIQUE_ID_OF_DIV"
              editorProps={{ $blockScrolling: true }}
              showPrintMargin={false}
              value={this.props.contents}
              style={{ border: '1px solid gold' }}
              wrapEnabled={Boolean(true)}
            />
          {/*</Resizable>*/}
        </Flexbox>
      </Flexbox>
    );
  }
}

export default Editor;
