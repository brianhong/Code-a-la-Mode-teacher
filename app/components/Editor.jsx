// @flow
import React, { Component } from 'react';
import brace from 'brace';
import { ipcRenderer } from 'electron';
import Flexbox from 'flexbox-react';
// import Resizable from 'react-resizable-box';
import AceEditor from 'react-ace';
import SplitPane from 'react-split-pane';

import Paper from 'material-ui/Paper';
import { Tab, Tabs } from 'material-ui/Tabs';

import 'brace/mode/javascript';
import 'brace/theme/solarized_dark';
import 'brace/ext/searchbox';
// import styles from './Home.css';
import FiletreeContainer from '../containers/FiletreeContainer';

import RightPanelContainer from '../containers/RightPanelContainer';
import EditorNavContainer from '../containers/EditorNavContainer';

import { getLastFromPath } from '../utils/file-functions';


/*
  Error with Webpack and how split is exported out, using require instead
  Keeping it here in case we might use it later

  const AceSplitEditor = require('react-ace').split;
*/
const style = {
  margin: 12,
};

type nextPropsType = {
  contents: Array<string>,
  repositoryPath: string,
  changeEditor: () => void,
  storageLogin: () => Object,
  selectedFileIndex: number,
  currEditorVal?: string
};

class Editor extends Component {
  props: {
    contents: Array<string>,
    repositoryPath: string,
    changeEditor: () => void,
    storageLogin: () => Object,
    currentOpenFiles: Array<string>,
    selectedFileIndex: number,
    loadFileFromTab: () => void,
    currEditorVal?: string
  };

  constructor(props) {
    super(props);
    this.state = { open: false, editorSize: '30vw', data: '', data2: ''};
  }

  componentWillMount() {
    const token = window.localStorage.getItem('token');
    const username = window.localStorage.getItem('username');
    if (token && username) return this.props.storageLogin(token, username); // eslint-ignore-line
  }

  componentWillReceiveProps(nextProps: nextPropsType) {

    ipcRenderer.send('editor-changes', nextProps.currEditorVal);
  }
  handleToggle = () => this.setState({ open: !this.state.open });
  handleResize = (size) => this.setState({ editorSize: size });

  render() {
    const { changeEditor, contents, currentOpenFiles, selectedFileIndex, repositoryPath, loadFileFromTab } = this.props;

    return (
      <div>
        <RightPanelContainer />
        <Flexbox flexDirection="row" minHeight="100vh" flexWrap="wrap" alignContent="flex-start">
          <EditorNavContainer />
          <Flexbox element="header" height="70px" width="100vw">
          </Flexbox>

          <Flexbox flexDirection="row" justifyContent="space-around">
            <div position="relative">
              <SplitPane split="vertical" defaultSize="240" onChange={this.handleResize} >
                <Paper style={style} zDepth={2}>
                  <FiletreeContainer directory={'/'} socket={this.socket} />
                </Paper>
                  <Paper style={style} zDepth={2} >
                    <Tabs value={selectedFileIndex} >
                      {
                        currentOpenFiles && currentOpenFiles.map((filePath, index) => (
                          <Tab
                            key={filePath}
                            label={getLastFromPath(filePath)}
                            value={index}
                            id={filePath} // TODO: Preferably not on id but this stops throwing an error for now
                            onActive={(tab) => loadFileFromTab(tab.props.id, currentOpenFiles, contents)}
                          />
                        ))
                      }
                    </Tabs>
                    <AceEditor
                      value={contents[selectedFileIndex]}
                      mode="javascript"
                      theme="solarized_dark"
                      width={'100%'}
                      onChange={(newValue, event) => { changeEditor(newValue, selectedFileIndex, contents)} }
                      name="UNIQUE_ID_OF_DIV"
                      wrapEnabled={true}
                      editorProps={{ $blockScrolling: Infinity }}
                    />
                  </Paper>
              </SplitPane>
            </div>
          </Flexbox>
        </Flexbox>
      </div>
    );
  }
}

export default Editor;