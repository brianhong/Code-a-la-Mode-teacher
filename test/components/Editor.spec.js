import React from 'react';
import { shallow } from 'enzyme';
import AceEditor from 'react-ace';
import renderer from 'react-test-renderer';
import Editor from '../../app/components/Editor';

import changeEditorImport from '../../app/actions/editor-actions';

describe('Editor component', () => {

  const component = shallow(<Editor changeEditor={changeEditorImport} />);

  it('should receive a changeEditor function in props', () => {
    expect(component.instance().props.changeEditor).toBe(changeEditorImport);
  });

  it('should display the AceEditor', () => {
    const tree = renderer
      .create(
        <div>
          <AceEditor
            // mode="javascript"
            // theme="solarized_dark"
            // onChange={changeEditor}
            // name="UNIQUE_ID_OF_DIV"
            // editorProps={{ $blockScrolling: true }}
          />
        </div>
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
