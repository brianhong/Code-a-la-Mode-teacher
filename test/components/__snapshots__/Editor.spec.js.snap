// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor component should display the AceEditor 1`] = `
<div>
  <div
    id="brace-editor"
    style={
      Object {
        "height": "500px",
        "width": "500px",
      }
    }
  />
</div>
`;
